superduperdb/components/model.py

```diff
@@ -735,144 +733,35 @@ def validate(
-        metrics: t.Optional[t.Sequence[Metric]] = None,
-        select: t.Optional[Select] = None,
-        validation_sets: t.Optional[t.Sequence[t.Union[str, Dataset]]] = None,
-        **kwargs,
-    ) -> t.Optional[Pipeline]:
-        """
-        Fit the model on the given data.
-
-        :param X: The key of the input data to use for training
-        :param y: The key of the target data to use for training
-        :param configuration: The training configuration (optional)
-        :param data_prefetch: Whether to prefetch the data (optional)
-        :param db: The datalayer (optional)
-        :param dependencies: The dependencies (optional)
-        :param metrics: The metrics to evaluate on (optional)
-        :param select: The select to use for training (optional)
-        :param validation_sets: The validation ``Dataset`` instances to use (optional)
-        """
-        if isinstance(select, dict):
-            # TODO replace with Document.decode(select)
-            select = Serializable.from_dict(select)
-
-        if validation_sets:
-            from superduperdb.components.dataset import Dataset
-
-            validation_sets = list(validation_sets)
-            for i, vs in enumerate(validation_sets):
-                if isinstance(vs, Dataset):
-                    assert db is not None
-                    db.add(vs)
-                    validation_sets[i] = vs.identifier
-
-        self.training_configuration = configuration or self.training_configuration
+    def _multiprocessing_wrapper(self, data):
+        args, kwargs = self.handle_input_type(data, self.signature)
+        return self.object(*args, **kwargs)
 
-        if db is not None:
-            db.add(self)
+    def predict_one(self, *args, **kwargs):
+        return self.object(*args, **kwargs)
 
-        if db is not None and db.compute.type == 'distributed':
-            return self.create_fit_job(
-                X,
-                select=select,
-                y=y,
-                **kwargs,
-            )(db=db, dependencies=dependencies)
+    def predict(self, dataset: t.Union[t.List, QueryDataset]) -> t.List:
+        outputs = []
+        if self.num_workers:
+            pool = multiprocessing.Pool(processes=self.num_workers)
+            for r in pool.map(
+                self._multiprocessing_wrapper, dataset  # type: ignore[arg-type]
+            ):
+                outputs.append(r)
+            pool.close()
+            pool.join()
         else:
-            return self._fit(
-                X,
-                y=y,
-                configuration=configuration,
-                data_prefetch=data_prefetch,
-                db=db,
-                metrics=metrics,
-                select=select,
-                validation_sets=validation_sets,
-                **kwargs,
-            )
+            for i in range(len(dataset)):
+                outputs.append(self._multiprocessing_wrapper(dataset[i]))
```

Comment:
its in else block right so no multiprocessing?

Reply:
Yes the name is not great.